ext/soap/soap.c

```diff
@@ -2368,6 +2368,8 @@ PHP_METHOD(SoapClient, SoapClient)
 				Z_TYPE_P(tmp) == IS_RESOURCE) {
 			context = php_stream_context_from_zval(tmp, 1);
 			Z_ADDREF_P(tmp);
+		} else {
+			context = php_stream_context_alloc(); 
```

Comment:
Without it, the stream in `http_connect` is created without a context, and the call to `php_stream_context_set_option` segfaults with a null pointer dereference.